src/pip/_internal/index.py

```diff
@@ -688,154 +677,33 @@ def egg_info_matches(
         like foo-2-2 which might be foo, 2-2 or foo-2, 2.
     :param link: The link the string came from, for logging on failure.
     """
-    match = _egg_info_re.search(egg_info)
-    if not match:
+    kwargs = {}
+    if _egg_info_re is not None:
+        kwargs["_egg_info_re"] = _egg_info_re
+    try:
+        return match_egg_info_version(egg_info, search_name, **kwargs)
+    except ValueError:
         logger.debug('Could not parse version from link: %s', link)
-        return None
-    if search_name is None:
-        full_match = match.group(0)
-        return full_match.split('-', 1)[-1]
-    name = match.group(0).lower()
-    # To match the "safe" name that pkg_resources creates:
-    name = name.replace('_', '-')
-    # project name and version must be separated by a dash
-    look_for = search_name.lower() + "-"
-    if name.startswith(look_for):
-        return match.group(0)[len(look_for):]
-    else:
-        return None
-
-
-class HTMLPage(object):
-    """Represents one page, along with its URL"""
-
-    def __init__(self, content, url, headers=None):
-        # Determine if we have any encoding information in our headers
-        encoding = None
-        if headers and "Content-Type" in headers:
-            content_type, params = cgi.parse_header(headers["Content-Type"])
-
-            if "charset" in params:
-                encoding = params['charset']
-
-        self.content = content
-        self.parsed = html5lib.parse(
-            self.content,
-            transport_encoding=encoding,
-            namespaceHTMLElements=False,
-        )
-        self.url = url
-        self.headers = headers
+    return None
 
-    def __str__(self):
-        return self.url
 
-    @classmethod
-    def get_page(cls, link, session=None):
-        if session is None:
-            raise TypeError(
-                "get_page() missing 1 required keyword argument: 'session'"
-            )
+def _handle_collect_links_fail(link, reason, url, meth=None):
+    if meth is None:
+        meth = logger.debug
+    meth("Could not fetch URL %s: %s - skipping", link, reason)
 
-        url = link.url
-        url = url.split('#', 1)[0]
 
-        # Check for VCS schemes that do not support lookup as web pages.
-        from pip._internal.vcs import VcsSupport
-        for scheme in VcsSupport.schemes:
-            if url.lower().startswith(scheme) and url[len(scheme)] in '+:':
-                logger.debug('Cannot look at %s URL %s', scheme, link)
-                return None
+def _iter_links(link, session):
```

Comment:
It looks like this is a 100+ line function. I would break it up some. For example, I think it would be good to at least separate the part making the request from the part that parses the content and loops over it. It looks like this separation perhaps existed even in the prior code?